src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -992,6 +993,233 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
+    otSrpClientService * srpService = nullptr;
+
+    VerifyOrExit(aInstanceName, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(aName, error = OT_ERROR_INVALID_ARGS);
+
+    // Check if service to remove exists.
+    for (otSrpClientService & service : mSrpServices)
+    {
+        if ((strcmp(service.mInstanceName, aInstanceName) == 0) && (strcmp(service.mName, aName) == 0))
+        {
+            srpService = &service;
+            break;
+        }
+    }
+
+    VerifyOrExit(srpService, error = OT_ERROR_NOT_FOUND);
+
+    error = otSrpClientRemoveService(mOTInst, srpService);
+
+exit:
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_StartSrpClient(const char * aHostName)
+{
+    otError error = OT_ERROR_NONE;
+    otIp6Address clientIpAddress;
+    otServiceConfig serverServiceConfig;
+
+    VerifyOrExit(aHostName, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(!otSrpClientIsRunning(mOTInst), error = OT_ERROR_ALREADY);
+
+    memcpy(&clientIpAddress, otThreadGetMeshLocalEid(mOTInst), sizeof(clientIpAddress));
+
+    error = otSrpClientSetHostName(mOTInst, aHostName);
+    VerifyOrExit(error == OT_ERROR_NONE, );
+
+    error = otSrpClientSetHostAddresses(mOTInst, &clientIpAddress, 1);
+    VerifyOrExit(error == OT_ERROR_NONE, );
+
+    VerifyOrExit(FindSrpServerService(serverServiceConfig), error = OT_ERROR_NOT_FOUND);
+
+    // Convert server service config data to get from it server IP address and port number.
+    otSockAddr serverSocketAddress;
+    serverSocketAddress.mAddress =
+        GenericThreadStackManagerImpl_OpenThread::ConvertSrpServerRloc16ToIpAddress(serverServiceConfig.mServerConfig.mRloc16);
+    serverSocketAddress.mPort =
+        GenericThreadStackManagerImpl_OpenThread::ConvertSrpServerDataToPort(serverServiceConfig.mServerConfig.mServerData);
+
+    error = otSrpClientStart(mOTInst, &serverSocketAddress);
+
+exit:
+    return MapOpenThreadError(error);
+}
+
+template <class ImplClass>
+bool GenericThreadStackManagerImpl_OpenThread<ImplClass>::FindSrpServerService(otServiceConfig & aServiceConfig)
+{
+    bool serviceFound = false;
+    Impl()->LockThreadStack();
+
+    otNetworkDataIterator iterator = OT_NETWORK_DATA_ITERATOR_INIT;
+
+    // Find in the server services the one used for SRP server purposes
+    while (otNetDataGetNextService(mOTInst, &iterator, &aServiceConfig) == OT_ERROR_NONE)
+    {
+        if (aServiceConfig.mEnterpriseNumber == kThreadServiceEnterpriseNumber &&
+            aServiceConfig.mServiceDataLength == sizeof(kSrpServiceId) && aServiceConfig.mServiceData[0] == kSrpServiceId)
+        {
+            serviceFound = true;
+            break;
+        }
+    }
+
+    Impl()->UnlockThreadStack();
+
+    return serviceFound;
+}
+
+template <class ImplClass>
+uint16_t GenericThreadStackManagerImpl_OpenThread<ImplClass>::ConvertSrpServerDataToPort(uint8_t * aServerData)
```

Comment:
`static_assert(OT_SERVER_DATA_MAX_SIZE >= 2, "");` to convince a reader that it's safe to access the array elements?

Reply:
This adds some more confidence to the reader, but anyway I believe we should check the value of `mServerDataLength`, and see if it's equal/equal_or_greater than two, probably not here but when you execute this function.

Reply:
Both changes done